torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -38,62 +38,64 @@ inline int64_t current_time_in_nanos() {
 } // namespace
 
 Reducer::Reducer(
-    std::vector<std::vector<torch::autograd::Variable>> inputs,
+    std::vector<std::vector<torch::autograd::Variable>> replicas,
+    std::vector<std::vector<size_t>> buckets,
```

Comment:
Good suggestion regarding the naming. Also renamed the `BucketIndex` struct and the corresponding vector to `VariableLocator`, as it locates a variable in the bucket structure, and it conflicted with the `bucket_indices` name as a vector of vector of indices.

Regarding `size_t`, it's common to use it for indexing, and it doesn't hurt. In practice we won't hit `UINT32_MAX` in this particular case of course, but I rather stick to using `size_t` everywhere instead of mix and match.